WalletWasabi/Crypto/ZeroKnowledge/ProofSystem.cs

```diff
@@ -0,0 +1,32 @@
+using NBitcoin.Secp256k1;
+using System.Linq;
+using System.Text;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Crypto.ZeroKnowledge.LinearRelation;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public static class ProofSystem
+	{
+		private static GroupElement Inf = GroupElement.Infinity;
+		private static Transcript Transcript => new Transcript(Encoding.UTF8.GetBytes("proof-of-parameters"));
+
+		private static GroupElementVector[] ProofOfParametersGenerators(GroupElement U, GroupElement Ma, Scalar t) =>
+			new GroupElementVector[]
+			{
+				new GroupElementVector( Generators.Gw, Generators.Gwp, Inf, Inf, Inf, Inf ),
+				new GroupElementVector( Inf, Inf, Generators.Gx0, Generators.Gx1, Inf, Generators.Ga ),
+				new GroupElementVector( Generators.Gw, Inf, U, t*U, Inf, Ma )
```

Comment:
5th index, witness component `t` is all `Inf`, so has no effect, see above comment

Reply:
maybe in a comment before line 17, we could just indicate which witness corresponds to which generators or public points. That way would be a little bit easier to map witnesses to generators.

Reply:
Done.